### ISSUE TITLE
non-free asio files

### ISSUE THREAD

**Issue by fsateler** [thestk/stk#60]
At least src/include/asio.{h,cpp} are part of the steinberg sdk, which is licensed under quite strict terms:

```

1.1
The object of this agreement consists of the Steinberg ASIO (Audio
Stream I/O) Software Development Kit version 2.3, respectively any
preview version of this Kit, comprising of documentation, example code,
and several ASIO examples. These are described hereinafter as the
"Licensed Software Developer Kit".

The license goes on to include other terms which seem to prohibit
modification or redistribution in most cases:

2.3
The Licensee has no permission to sell, license, give-away and/or
distribute the Licensed Software Developer Kit or parts of it for the
use as software developer kit in any way, for example, integrated in
any framework application, on any medium, including the Internet, to
any other person, including sub-licensors of the Licensee or companies
where the Licensee has any involvement. This includes re-working this
specification, or reverse-engineering any products based upon this
specification.

2.4
In case the Licensed Software Developer Kit is modified, it shall not
be published, sold or distributed without agreement of Steinberg.
Furthermore, the naming of the Licensed Software Developer Kit shall
not include “ASIO 2.3 SDK” or any combination containing the VST brand
without permission of Steinberg.
```

The files should probably be stripped from the stk sources.

**Comment by arielelkin** [comment]
Clause 2.1 of the [Steinberg ASIO SDK Licensing Agreement](http://www.steinberg.net/en/company/developers.html) stipulates:

> Steinberg hereby grants to the Licensee a non-exclusive, worldwide, nontransferable license during the term of this agreement to use the ASIO Interface Technology solely:
a) for the development of ASIO Drivers and/or for the development of an application that can host ASIO Drivers (herein ASIO Driver Compliant Product),
b) publish, sell or otherwise distribute a ASIO Driver Compliant Product under his own brand name that is using parts or all of the Licensed Software Developer Kit.

I believe we're doing `a`, and it seems like this clause grants us the right to do so. 

So Clause 2.3 really looks like a direct contradiction. 

[The problems created by that restriction were already raised over at the rtaudio repository](https://github.com/thestk/rtaudio/issues/16), @garyscavone said the following:
> Technically speaking, we are not supposed to distribute the ASIO SDK files.  That said, I find it silly that I am not supposed to bundle something that Steinberg makes available to developers for free.  Thus, I ignore that issue to save users the trouble of having to download the SDK themselves.  

And I'd agree. 

At any rate, Steinberg's intentions aren't very clear (to me at least), but the fact is that they open-sourced their ASIO files, so I don't see why us incorporating them here would upset them in any way. 

Thoughts?

**Comment by fsateler** [comment]
For my particular use case, all that I need is that the headers not be shipped with STK (AFAICT, ASIO is only useful on windows). Perhaps a make rule to download and extract the SDK would be an agreeable compromise?

**Comment by arielelkin** [comment]
>Perhaps a make rule to download and extract the SDK would be an agreeable compromise?

Sadly no, because Steinberg's not using any dependency manager for [their SDK](http://www.steinberg.net/en/company/developers.html). They're not even offering a git repository, so submodules are ruled out too. 

It's an awkward state of affairs but I can live it (in fact, we have lived with it for several years without, to my knowledge, any material damage being caused to Steinberg (whereas the contrary is true, as we've promoted usage and support of their protocol)). 

So I'd say we park this.
